src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+    - additional_information (optional, free-form text)
+
+    TODO: should we allow more fields in the input JSON file that are just ignored, so that people don't need to manually remove those fields in their dataset JSON file before calling this function?
+    TODO: should we support other serialization formats like pickle, Parquet, HDF5, etc.?
+    TODO: should we support another Python function calling this function, thus needing to keep both input and output in memory (for perf reason) instead of being serialized to disk?
+    TODO: `n_to_print` should be controlled by LOG_LEVEL. Also probably shouldn't print markdown stuff in CLI.
+    TODO: should we allow customizing which model to use for rubric_generator (right now it's hardcoded to "gpt4_CoT_v0")?
+    """
+    print(f"Calling get_rubrics! input_path: {input_path}")
+    assert input_path.endswith(".json"), "only JSON format is supported"
+    df = pd.read_json(input_path)
+    check_df_fields(
+        df, 
+        required_fields={"prompt"},
+        optional_fields={"category", "additional_information"},
+    )
+    # Fill in missing fields
+    for col in ["category", "additional_information"]:
+        if col not in df.columns:
+            df[col] = ""
+    rubric_generator_name = "gpt4_CoT_v0"
+
+    rubric_brainstormer = RubricBrainstormer(annotators_config=rubric_generator_name)
+    criteria = rubric_brainstormer(df)
+    df_criteria = rubric_brainstormer.make_df_rubrics(criteria)
+
+    # TODO: this should probably be included in the post-processing of the rubric_brainstormer
+    # copy "prompt" column to "final_prompt" column
+    df_criteria["final_prompt"] = df_criteria["prompt"]
+    
+    df_rubrics = get_detailed_rubrics(df_criteria, n_to_print=3, is_store_missing_annotations=True, annotators_config=rubric_generator_name)
+    df_rubrics = cleanup_df_rubrics(df_rubrics)
+    df_rubrics.to_json(output_path, orient='records', indent=4)
+
+
+def get_completions(
+    model_configs: List[str] = [],
+    input_path: str = "instructions_with_rubrics.json",
+    output_path_prefix: str = f"output_"
+):
+    """
+    For each model, we will save the completion to {output_path_prefix}{model_config}.json file.
+    """
+    print(f"Calling get_outputs! model_configs: {model_configs}")
+    # model_configs = string_to_list(model_configs)
+    assert input_path.endswith(".json"), "only JSON format is supported"
+    df_rubrics = pd.read_json(input_path)
+    check_df_fields(
+        df_rubrics,
+        # TODO: can we dedup "prompt" and "final_prompt"?
+        required_fields={"prompt", "final_prompt", "category", "additional_information", "raw_completion", "scoring_scales", "criteria", "detailed_analytic_rubric"},
+        # TODO: do we require any of these fields?
+        optional_fields={'annotator', 'reference_answer', 'critique', 'time_per_example', 'clear_goals', 'price_per_example'},
+    )
+    # TODO: parallelize this?
+    for model_config in model_configs:
```

Comment:
if we parallelize then I agree that having a list of model_configs is nice. The issue with parallelization is that it won't work well with the current caching mechanism in alpaca_eval. Changing that would require a good amount of work